atlascope/core/migrations/0003_dataset_importers.py

```diff
@@ -0,0 +1,18 @@
+# Generated by Django 4.0 on 2021-12-07 20:13
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('core', '0002_base_models'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='dataset',
+            name='importer',
+            field=models.CharField(choices=[], max_length=100, null=True),
```

Comment:
This migration says that the choice list is blank. Will that cause problems if we add or otherwise change the list of available importers at runtime? Or is that simply how choice-based Django model field work?

Reply:
This part I am not so sure about. It worked fine for my setup, but I'm wondering if it will break on someone else's. If it does, we may have to find a complicated workaround, since each instance will have a different list of installed plugins. I think we will be safe, since most of the time, changes to the choices field don't warrant a full migration. Still, could you spin it up and give it a try?

Reply:
Instead of dealing with possible issues on this choice field, @waxlamp pointed out that a validator function would be far easier, since its logic is done at runtime. This is a very elegant implementation that I had surprisingly never used! This change is made in ee3995f.

Reply:
The change looks great. It's more explicit, and it avoids the semantic constraints inherent in the idea of the choice field.